RDMAUpdate/main/Utils/HandlerUtil.py

```diff
@@ -148,13 +149,14 @@ def _parse_config(self, ctxt):
                 thumb = handlerSettings['protectedSettingsCertThumbprint']
                 cert = waagent.LibDir + '/' + thumb + '.crt'
                 pkey = waagent.LibDir + '/' + thumb + '.prv'
-                waagent.SetFileContents('/tmp/kk', protectedSettings)
-                cleartxt = None
-                cleartxt = waagent.RunGetOutput("base64 -d /tmp/kk | openssl smime  -inform DER -decrypt -recip " + cert + "  -inkey " + pkey)[1]
-                os.remove("/tmp/kk")
-                if cleartxt == None:
-                    self.error("OpenSSh decode error using  thumbprint " + thumb)
-                    do_exit(1,operation,'error','1', operation + ' Failed')
+                f = tempfile.NamedTemporaryFile(delete=False)
+                f.close()
+                waagent.SetFileContents(f.name, protectedSettings)
+                cleartxt = waagent.RunGetOutput("base64 -d " + f.name + " | openssl smime  -inform DER -decrypt -recip " + cert + "  -inkey " + pkey)[1]
```

Comment:
I'm wrong. Hosung fixed this in December 2016. The commit in the azure-Linux-extensions repo is [7315f17d](https://github.com/Azure/azure-linux-extensions/commit/7315f17d4616b5220749ba4b8cb011f9a55c0cb7#diff-7f84beab04579bef70043ca0cc72fb85). The change completely eliminates the use of the temporary file.